Automattic/wp-calypso
# Stats: Extended number component with default formatter

#### Proposed Changes

* adding a default number formatter to Calypso components
* importing the formatter in Stats components and replacing it in places where it was used previously

#### Testing Instructions



* no visual changes should be present
* go to Stats page and check that the numbers on the page look correctly for your region
* Examples of affected places: components with the horizontal bar, All-time highlights section on the Insights page

#### Pre-merge Checklist





Related to #71639
